launch/launch/frontend/parse_substitution.py

```diff
@@ -93,14 +93,21 @@ def template(self, fragments):
     double_quoted_template = template
 
 
-grammar_file = os.path.join(get_package_share_directory('launch'), 'frontend', 'grammar.lark')
-parser = Lark.open(grammar_file, start='template')
```

Comment:
Does anyone know why `Load.open()` was chosen here instead of the more intuitive constructor `Lark()`?

Reply:
It was a completely random choice. I'm not sure which one is more intuitive.

Reply:
The constructor is imo the preferred way since it allows arbitrary options to be passed. In newer versions that includes using a cached state of the parsed grammar.

Reply:
If you want to make that change here, it sounds good to me too.

Reply:
Done in fc22b48347e3e6b8e575b2f5ec3a6a932abccbaf.

Reply:
The constructor takes the actual grammar string rather than a path. I modified it in 1210d0ac93aca05977a8faaa3605fe9a99d1431a which also nicely shows that the file is being closed after parsing the grammar.